src/commonMain/kotlin/com/mattprecious/stacker/stack/RealStackManager.kt

```diff
@@ -90,42 +78,11 @@ class RealStackManager(
 		)
 	}
 
-	private fun getTree(): Tree {
-		val branches = branchQueries.selectAll().executeAsList()
-		val treeBranches = branches.map { TreeBranch(it.name, it.parentSha) }.associateBy { it.name }
-		return Tree(
-			root = treeBranches[branches.single { it.parent == null }.name],
-			parents = branches.associateBy({ it.name }, { treeBranches[it.parent] }),
-			children = branches.groupBy({ it.parent }, { treeBranches[it.name]!! }),
+	private fun getTree(): TreeNode<Branch>? {
+		return treeOf(
+			elements = branchQueries.selectAll().executeAsList(),
```

Comment:
Especially since you create the tree one or more times in a single command, all the more reason to make it super fast.

Reply:
Yeah, for sure. But even so I should probably take a pass over everything and try to bring my trees-per-command down to 1. `getBranch()` should probably just be an extension function on `TreeNode` so I'm encouraged to only grab the tree once.